src/details/ArborX_Ray.hpp

```diff
@@ -273,6 +274,45 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   return (tmax - tmin);
 }
 
+// Möller–Trumbore intersection algorithm
+KOKKOS_INLINE_FUNCTION bool intersects(Ray const &ray, Triangle const &triangle)
+{
+  auto ab = makeVector(triangle.a, triangle.b);
+  auto ac = makeVector(triangle.a, triangle.c);
+
+  auto h = crossProduct(ray.direction(), ac);
+  auto a = dotProduct(ab, h);
+
+  auto const epsilon = 0.0000001f;
+  if (a > -epsilon && a < epsilon)
+  {
+    return false; // ray parallel to the triangle
+  }
+
+  auto f = 1.f / a;
+  auto s = makeVector(triangle.a, ray.origin());
+  auto u = f * dotProduct(s, h);
+  if (u < 0.f || u > 1.f)
+  {
+    return false;
+  }
+
+  auto q = crossProduct(s, ab);
+  auto v = f * dotProduct(ray.direction(), q);
+  if (v < 0.f || u + v > 1.f)
+  {
+    return false;
+  }
+
+  auto t = f * dotProduct(ac, q);
```

Comment:
Is there a way to have access to this variable? It's the distance between the origin point and the intersection with the triangle. So we can simply get the intersection point using `origin + t * ray.direction()`.

Reply:
What would you want to set it to for any of the early returns above?

Reply:
I guess it would make sense to set that to infty since the line never intersects.